build-js.sh

```diff
@@ -77,872 +75,46 @@ python3 $BINARYEN_SCRIPTS/embedwast.py $BINARYEN_SRC/passes/wasm-intrinsics.wast
 
 echo "compiling source files"
 
-mkdir -p ${OUT}
+# internal source files or directories to exclude
+EXCLUDE=(
+  "$BINARYEN_SRC/support/archive.cpp"
+  "$BINARYEN_SRC/support/command-line.cpp"
+  "$BINARYEN_SRC/support/path.cpp"
+  "$BINARYEN_SRC/tools/"
+)
+
+# external source files to include
+INCLUDE=()
+
+readarray -d '' -t SOURCES < <(find $BINARYEN_SRC -name "*.cpp" -print0 | sort -z)
+EXCLUDE+=("$BINARYEN_SRC/binaryen-c.cpp")
```

Comment:
Why this line?   Isn't this the main file we do want to include?

Reply:
Yeah, but the file is excluded when compiling `shared.o`, and only included in the final step.